parser.cs

```diff
@@ -449,86 +449,479 @@ public class Parser {
+                factor();
+                factorTail();
+                break;
+            default:
+                error("An error with term");
+                break;
+        }
     }
 
     private void factorTail(){
-
+        switch(__lookahead.Type){
+            case TOKENS.AND:
+            case TOKENS.DIV:
+            case TOKENS.MOD:
+            case TOKENS.FLOAT_DIVIDE:
+            case TOKENS.TIMES:
+                // Rule 92
+                multiplyingOperator();
+                factor();
+                factorTail();
+                break;
+            case TOKENS.DO:
+            case TOKENS.DOWNTO:
+            case TOKENS.END:
+            case TOKENS.OR:
+            case TOKENS.THEN:
+            case TOKENS.TO:
+            case TOKENS.COMMA:
+            case TOKENS.EQUAL:
+            case TOKENS.GEQUAL:
+            case TOKENS.GTHAN:
+            case TOKENS.LEQUAL:
+            case TOKENS.LTHAN:
+            case TOKENS.MINUS:
+            case TOKENS.NEQUAL:
+            case TOKENS.PLUS:
+            case TOKENS.RPAREN:
+            case TOKENS.SCOLON:
+                // Rule 93
+                break;
+            default:
+                error("An error with factorTail");
+                break;
+        }
     }
 
     private void multiplyingOperator(){
-
+        switch(__lookahead.Type){
+            case TOKENS.AND:
+                // Rule 98
+                match(TOKENS.AND);
+                break;
+            case TOKENS.DIV:
+                // Rule 96
+                match(TOKENS.DIV);
+                break;
+            case TOKENS.MOD:
+                //Rule 97
+                match(TOKENS.MOD);
+                break;
+            case TOKENS.FLOAT_DIVIDE:
+                // Rule 95
+                match(TOKENS.FLOAT_DIVIDE);
+                break;
+            case TOKENS.TIMES:
+                // Rule 94
+                match(TOKENS.TIMES);
+                break;
+            default:
+                error("An error with multiplyingOperator");
+                break;
+        }
     }
 
     private void factor(){
-
+        switch(__lookahead.Type){
+            case TOKENS.FALSE:
+                // Rule 103
+                match(TOKENS.FALSE);
+                break;
+            case TOKENS.NOT:
+                // Rule 104
+                match(TOKENS.NOT);
+                factor();
+                break;
+            case TOKENS.TRUE:
+                // Rule 102
+                match(TOKENS.TRUE);
+                break;
+            case TOKENS.IDENTIFIER:
+                // Rule 106
+                functionIdentifier();
+                optionalActualParameterList();
+                break;
+            case TOKENS.INTEGER_LIT:
+                // unsigned integer? Rule 99?
+                match(TOKENS.INTEGER_LIT);
+                break;
+            case TOKENS.FIXED_LIT:
+                // Question mark on table? rule 100? not sure
```

Comment:
Oh, so for some reason the grammar doesn't specify that factor should go to fixed lit, but it totes m'goats should. So just match the token that is in the case for float and fixed and it should be fine.